### PR TITLE
ta_on_candle (not loop, with optional flag in config.json) Resubmitting - because GIT.

### PR DESCRIPTION
**Pull request by creslinux**
This is the last cut that was in #1117 before i closed that PR

This PR allows a user to set the flag `ta_on_candle` in their config.json

This will change the behaviour of the the bot to only process indicators
when there is a new candle to be processed for that pair.

the problem being addressed is in a 5 minute candle the same dataframe with no new 
candle data will be processed by TA indicators looking at only candle data 100 times. 
In a 4 hour timeframe this is many thousands of if times. 

the bot then has to deal with expensive CPU load slowing the system and ignoring thouands of false buy/sell signals. 

The new behavour is optional, the default is no change in current behaviour. 

The test is against a dict of  {'pair': candle last processed date} not matching current last candle row OR `ta_on_candle` no being true

Either test will execute existing behaviour, to process TA regardless.

### CODE REVIEW

**Comment by creslinux** [comment]
pulling - needs more work on the logic.

**Comment by creslinux** [comment]
This PR is failing with metadata is a str with no  get attribute. 
Looking at metadata, its a dict, and get attribute is working as expected. 

Ive tried to find an offended test, but no look. 

```
>       pair = metadata.get('pair')
E       AttributeError: 'str' object has no attribute 'get'
```

``` print(type(metadata))
        pair = metadata.get('pair')
        print(metadata)
        print(pair)
```
produces: 
```
<class 'dict'>
{'pair': 'XLM/BTC'}
XLM/BTC
```

**Comment by xmatthias** [comment]
i think there is something broken with the tests - metadata was never used in the tests apart from passing it on - so i guess the tests call it with a str instead of a dict...

i can have a look at it tomorrow i guess - (as a separate pr as it's not related to this at all)

**Comment by xmatthias** [comment]
#1130 should fix the failing test.

**Comment by xmatthias** [comment]
added some (hopefully) improvements in a PR to your branch. https://github.com/creslinux/freqtrade/pull/2

mainly and most importantly - make this configurable from the strategy - that's the only place this setting makes sense to me - but config can override it

File: docs/configuration.md [hunk 1]
```diff
@@ -22,7 +22,8 @@ The table below will list all configuration parameters.
 | `stake_amount` | 0.05 | Yes | Amount of crypto-currency your bot will use for each trade. Per default, the bot will use (0.05 BTC x 3) = 0.15 BTC in total will be always engaged. Set it to 'unlimited' to allow the bot to use all avaliable balance.
 | `ticker_interval` | [1m, 5m, 30m, 1h, 1d] | No | The ticker interval to use (1min, 5 min, 30 min, 1 hour or 1 day). Default is 5 minutes
 | `fiat_display_currency` | USD | Yes | Fiat currency used to show your profits. More information below. 
-| `dry_run` | true | Yes | Define if the bot must be in Dry-run or production mode. 
+| `dry_run` | true | Yes | Define if the bot must be in Dry-run or production mode.
+| `ta_on_candle` | false | No | If set to true indicators are processed only once a new candle arrives. If false each loop populates the indicators, this will mean the same candle is processed many times creating system load but can be useful of your strategy depends on tick data not only candle. Can be set either in Configuration or in the strategy.
```
[the remark this answers is not in the document]
Comment (xmatthias) [reply docs/configuration.md]: good name - thanks!

**Comment by xmatthias** [comment]
Good comments - addressed them all, please have a look again.

A note on your general comment:
there is no need to process 5m Candles every 5 seconds or so, they will not change and don't include partial candles.

Once "live"-ticker is available to the strategy, this may change for some strategies - but the decision is left to the strategy (that's why i moved the main decision criteria there).

This is not the end of this (not at all) - once we have this - and async - we should also look at reducing the time for "get_ticker" - either by asyncing this too - or by calling all pairs in one call and then passing that around.

**Comment by mishaker** [comment]
Looks ok to me 👍

**Comment by nullart2** [comment]
nice PR... this is ok for me 👍

**Pull request merged**